src/e2e-test/features/sfmcsink/DesignTimeWithMacros.feature

```diff
@@ -63,3 +63,15 @@ Feature: Salesforce Marketing Cloud Sink - Design time scenarios (macros)
     And Click on the Macro button of Property: "maxBatchSize" and set the value to: "maxBatchSize"
     And Click on the Macro button of Property: "columnMapping" and set the value to: "columnMapping"
     Then Validate "Salesforce Marketing" plugin properties
+
+  @BATCH-TS-SFMC-DSGN-MACRO-04
+  Scenario: Verify user should be able to validate the plugin when configured for Connection Manager property with macros
+    When Open Datafusion Project to configure pipeline
+    And Select plugin: "Salesforce Marketing" from the plugins list as: "Source"
+    And Navigate to the properties page of plugin: "Salesforce Marketing"
+    And Enter input plugin property: "referenceName" with value: "Referencename"
+    And Click plugin property: "switch-useConnection"
+    And Click on the Macro button of Property: "Connection" and set the value to: "Connection"
+    And Click on the Macro button of Property: "dataExtension" and set the value to: "dataExtension"
```

Comment:
This Step looks wrong there is no data-cy as dataExtension it should be Object if you are using data extension object. Inspect once on the field.